### ISSUE TITLE
Attention Mask for T5 / T2V Attention

### ISSUE THREAD

**Issue by karan-dalal** [THUDM/CogVideo#686]
### System Info / 系統信息

N/A

### Information / 问题信息

- [x] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

N/A

### Expected behavior / 期待表现

Hi. I have a few questions on the way text was integrated in the T2V model.

### **T5 Model**

When producing the text embeddings, you pad the tokens to the `self.max_length` first. Then, you forward through the T5 encoder:
https://github.com/THUDM/CogVideo/blob/bbe909d7f73d3cee166653f867a824c1b2151402/sat/sgm/modules/encoders/modules.py#L276

How come you don't include an attention mask? Right now, the T5 encoder is able to attend to padded tokens during the encoding process.

The better thing to do would be:
`outputs = model(input_ids=inputs['input_ids'], attention_mask=inputs['attention_mask'])`

### **Attention in DiT**

When training the DiT, you concat the text to the video embedding and pass it through attention. But your attention mask is set to all ones:
https://github.com/THUDM/CogVideo/blob/bbe909d7f73d3cee166653f867a824c1b2151402/sat/dit_video_concat.py#L855

Because your text is right padded, you are doing attention over `[Text, Pad Tokens, Video Embedding]` – why do you not use an attention mask?
